app/src/main/java/org/mifos/mobile/ui/fragments/UpdatePasswordFragment.kt

```diff
@@ -63,7 +65,13 @@ class UpdatePasswordFragment : BaseFragment(), TextWatcher, OnFocusChangeListene
 
                 is RegistrationUiState.Error -> {
                     hideProgress()
-                    showError(state.exception)
+                    if (TextUtils.isEmpty(state.exception)) {
+                        showError(getString(R.string.not_found_error))
+                        return@observe
+                    }
+                    val errorResponseObject = JSONObject(state.exception)
+                    val error = errorResponseObject.getString("error")
+                    showError(error)
```

Comment:
shouldn't this logic be moved to the viewmodel?  the general principle of MVVM is to keep the business logic and data manipulations in the ViewModel, while the View (Fragment in this case) should focus on displaying the data and handling user interactions.